developer/src/kmc-package/src/compiler/messages.ts

```diff
@@ -110,5 +110,9 @@ export class CompilerMessages {
   static Error_PackageMustContainAModelOrAKeyboard = () => m(this.ERROR_PackageMustContainAModelOrAKeyboard,
     `Package must contain a lexical model or a keyboard.`);
   static ERROR_PackageMustContainAModelOrAKeyboard = SevError | 0x0019;
+
+  static Warn_JsKeyboardFileIsMissing = (o:{id: string}) => m(this.WARN_JsKeyboardFileIsMissing,
+    `Keyboard ${o.id} targets touch devices but corresponding ${o.id}.js file is not in the package.`);
+  static WARN_JsKeyboardFileIsMissing = SevWarn | 0x001A;
```

Comment:
I was going to ask if there would also be a `Warn_JsModelFileIsMissing`, but I think that's handled by `Error_PackageMustContainAModelOrAKeyboard`.

Reply:
Yes, this is a test specifically for keyboards -- if the keyboard targets touch, then the package should include a .js.

Reply:
And does `Error_PackageMustContainAModelOrAKeyboard` handle if targets is `any` but the kmx isn't included in the package?

Reply:
The question doesn't really make sense -- `targets` data comes from the .kmx so it must already be in the package!

Reply:
gotcha. I was asking where the .kmn has targets `any`, but no kmx added to the .kps.